numpy/core/src/umath/ufunc_object.c

```diff
@@ -6238,49 +6281,11 @@ ufunc_at(PyUFuncObject *ufunc, PyObject *args)
         }
     }
 
-    /* 
-     * Create a map iterator (there is no multi-mapiter, so we need at least 2
-     * iterators: one for the mapping, and another for the second operand)
-     */
-    iter = (PyArrayMapIterObject *)PyArray_MapIterArrayCopyIfOverlap(
-        op1_array, idx, 1, op2_array);
-    if (iter == NULL) {
-        goto fail;
-    }
-    op1_array = iter->array;  /* May be updateifcopied on overlap */
-
-    if (op2_array != NULL) {
-        /*
-         * May need to swap axes so that second operand is
-         * iterated over correctly
-         */
-        if ((iter->subspace != NULL) && (iter->consec)) {
-            PyArray_MapIterSwapAxes(iter, &op2_array, 0);
-            if (op2_array == NULL) {
-                /* only on memory allocation failure */
-                goto fail;
-            }
-        }
-
-        /*
-         * Create array iter object for second operand that
-         * "matches" the map iter object for the first operand.
-         * Then we can just iterate over the first and second
-         * operands at the same time and not have to worry about
-         * picking the correct elements from each operand to apply
-         * the ufunc to.
-         */
-        if ((iter2 = (PyArrayIterObject *)\
-             PyArray_BroadcastToShape((PyObject *)op2_array,
-                                        iter->dimensions, iter->nd))==NULL) {
-            goto fail;
-        }
-    }
-
     PyArrayMethodObject *ufuncimpl = NULL;
-
     {
         /* Do all the dtype handling and find the correct ufuncimpl */
+        Py_INCREF(PyArray_DESCR(op1_array));
```

Comment:
I pushed a commit with those changes since that looked right and seemed to pass the test suite.  I think closer leak-checking may be relevant, but will prefer as a follow-up.